@psychocoderHPC I've tried updating to the version of mdspan that you suggested in #2605, but I got a lot of errors like
```
/builds/hzdr/crp/alpaka/build/_deps/mdspan-src/include/experimental/__p2630_bits/submdspan_mapping.hpp:276:34: error: reference to __host__ function '<deduction guide for tuple><std::integral_constant<unsigned long, 1>, std::integral_constant<unsigned long, 1>>' in __host__ __device__ function
  276 |                                  tuple{detail::stride_of(slices)...})),
      |                                  ^
```
in the CI.

Did I miss something ?

--- a/script/install_cuda.sh
+++ b/script/install_cuda.sh
@@ -41,6 +41,9 @@ else
         elif [[ "$(cat /etc/os-release)" == *"22.04"* ]]
         then
             ALPAKA_CUDA_DISTRO=ubuntu2204
+        elif [[ "$(cat /etc/os-release)" == *"24.04"* ]]
+        then
+            ALPAKA_CUDA_DISTRO=ubuntu2404
         fi
 
         # Set the correct CUDA downloads
@@ -89,8 +92,8 @@ else
         elif [ "${ALPAKA_CI_CUDA_VERSION}" == "12.0" ]
         then
             ALPAKA_CUDA_PKG_DEB_NAME=cuda-repo-"${ALPAKA_CUDA_DISTRO}"-12-0-local
-            ALPAKA_CUDA_PKG_FILE_NAME="${ALPAKA_CUDA_PKG_DEB_NAME}"_12.0.0-525.60.13-1_amd64.deb
-            ALPAKA_CUDA_PKG_FILE_PATH=https://developer.download.nvidia.com/compute/cuda/12.0.0/local_installers/${ALPAKA_CUDA_PKG_FILE_NAME}
+            ALPAKA_CUDA_PKG_FILE_NAME="${ALPAKA_CUDA_PKG_DEB_NAME}"_12.0.1-525.85.12-1_amd64.deb
+            ALPAKA_CUDA_PKG_FILE_PATH=https://developer.download.nvidia.com/compute/cuda/12.0.1/local_installers/${ALPAKA_CUDA_PKG_FILE_NAME}
             ALPAKA_CUDA_OLD_KEYS=false
         elif [ "${ALPAKA_CI_CUDA_VERSION}" == "12.1" ]
         then
@@ -101,8 +104,8 @@ else
         elif [ "${ALPAKA_CI_CUDA_VERSION}" == "12.2" ]
         then
             ALPAKA_CUDA_PKG_DEB_NAME=cuda-repo-"${ALPAKA_CUDA_DISTRO}"-12-2-local
-            ALPAKA_CUDA_PKG_FILE_NAME="${ALPAKA_CUDA_PKG_DEB_NAME}"_12.2.1-535.86.10-1_amd64.deb
-            ALPAKA_CUDA_PKG_FILE_PATH=https://developer.download.nvidia.com/compute/cuda/12.2.1/local_installers/${ALPAKA_CUDA_PKG_FILE_NAME}
+            ALPAKA_CUDA_PKG_FILE_NAME="${ALPAKA_CUDA_PKG_DEB_NAME}"_12.2.2-535.104.05-1_amd64.deb
+            ALPAKA_CUDA_PKG_FILE_PATH=https://developer.download.nvidia.com/compute/cuda/12.2.2/local_installers/${ALPAKA_CUDA_PKG_FILE_NAME}
             ALPAKA_CUDA_OLD_KEYS=false
         elif [ "${ALPAKA_CI_CUDA_VERSION}" == "12.3" ]
         then
@@ -113,14 +116,14 @@ else
         elif [ "${ALPAKA_CI_CUDA_VERSION}" == "12.4" ]
         then
             ALPAKA_CUDA_PKG_DEB_NAME=cuda-repo-"${ALPAKA_CUDA_DISTRO}"-12-4-local
-            ALPAKA_CUDA_PKG_FILE_NAME="${ALPAKA_CUDA_PKG_DEB_NAME}"_12.4.0-550.54.14-1_amd64.deb
-            ALPAKA_CUDA_PKG_FILE_PATH=https://developer.download.nvidia.com/compute/cuda/12.4.0/local_installers/${ALPAKA_CUDA_PKG_FILE_NAME}
+            ALPAKA_CUDA_PKG_FILE_NAME="${ALPAKA_CUDA_PKG_DEB_NAME}"_12.4.1-550.54.15-1_amd64.deb
+            ALPAKA_CUDA_PKG_FILE_PATH=https://developer.download.nvidia.com/compute/cuda/12.4.1/local_installers/${ALPAKA_CUDA_PKG_FILE_NAME}
             ALPAKA_CUDA_OLD_KEYS=false
         elif [ "${ALPAKA_CI_CUDA_VERSION}" == "12.5" ]
         then
             ALPAKA_CUDA_PKG_DEB_NAME=cuda-repo-"${ALPAKA_CUDA_DISTRO}"-12-5-local
-            ALPAKA_CUDA_PKG_FILE_NAME="${ALPAKA_CUDA_PKG_DEB_NAME}"_12.5.0-555.42.02-1_amd64.deb
-            ALPAKA_CUDA_PKG_FILE_PATH=https://developer.download.nvidia.com/compute/cuda/12.5.0/local_installers/${ALPAKA_CUDA_PKG_FILE_NAME}
+            ALPAKA_CUDA_PKG_FILE_NAME="${ALPAKA_CUDA_PKG_DEB_NAME}"_12.5.1-555.42.06-1_amd64.deb
+            ALPAKA_CUDA_PKG_FILE_PATH=https://developer.download.nvidia.com/compute/cuda/12.5.1/local_installers/${ALPAKA_CUDA_PKG_FILE_NAME}
             ALPAKA_CUDA_OLD_KEYS=false
         elif [ "${ALPAKA_CI_CUDA_VERSION}" == "12.6" ]
         then

--- a/script/install_cuda.sh
+++ b/script/install_cuda.sh
@@ -128,11 +128,29 @@ else
         elif [ "${ALPAKA_CI_CUDA_VERSION}" == "12.6" ]
         then
             ALPAKA_CUDA_PKG_DEB_NAME=cuda-repo-"${ALPAKA_CUDA_DISTRO}"-12-6-local
-            ALPAKA_CUDA_PKG_FILE_NAME="${ALPAKA_CUDA_PKG_DEB_NAME}"_12.6.1-560.35.03-1_amd64.deb
-            ALPAKA_CUDA_PKG_FILE_PATH=https://developer.download.nvidia.com/compute/cuda/12.6.1/local_installers/${ALPAKA_CUDA_PKG_FILE_NAME}
+            ALPAKA_CUDA_PKG_FILE_NAME="${ALPAKA_CUDA_PKG_DEB_NAME}"_12.6.3-560.35.05-1_amd64.deb
+            ALPAKA_CUDA_PKG_FILE_PATH=https://developer.download.nvidia.com/compute/cuda/12.6.3/local_installers/${ALPAKA_CUDA_PKG_FILE_NAME}
+            ALPAKA_CUDA_OLD_KEYS=false
+        elif [ "${ALPAKA_CI_CUDA_VERSION}" == "12.8" ]
+        then
+            ALPAKA_CUDA_PKG_DEB_NAME=cuda-repo-"${ALPAKA_CUDA_DISTRO}"-12-8-local
+            ALPAKA_CUDA_PKG_FILE_NAME="${ALPAKA_CUDA_PKG_DEB_NAME}"_12.8.1-570.124.06-1_amd64.deb
+            ALPAKA_CUDA_PKG_FILE_PATH=https://developer.download.nvidia.com/compute/cuda/12.8.1/local_installers/${ALPAKA_CUDA_PKG_FILE_NAME}
+            ALPAKA_CUDA_OLD_KEYS=false
+        elif [ "${ALPAKA_CI_CUDA_VERSION}" == "12.9" ]
+        then
+            ALPAKA_CUDA_PKG_DEB_NAME=cuda-repo-"${ALPAKA_CUDA_DISTRO}"-12-9-local
+            ALPAKA_CUDA_PKG_FILE_NAME="${ALPAKA_CUDA_PKG_DEB_NAME}"_12.9.1-575.57.08-1_amd64.deb
+            ALPAKA_CUDA_PKG_FILE_PATH=https://developer.download.nvidia.com/compute/cuda/12.9.1/local_installers/${ALPAKA_CUDA_PKG_FILE_NAME}
+            ALPAKA_CUDA_OLD_KEYS=false
+        elif [ "${ALPAKA_CI_CUDA_VERSION}" == "13.0" ]
+        then
+            ALPAKA_CUDA_PKG_DEB_NAME=cuda-repo-"${ALPAKA_CUDA_DISTRO}"-13-0-local
+            ALPAKA_CUDA_PKG_FILE_NAME="${ALPAKA_CUDA_PKG_DEB_NAME}"_13.0.2-580.95.05-1_amd64.deb
+            ALPAKA_CUDA_PKG_FILE_PATH=https://developer.download.nvidia.com/compute/cuda/13.0.2/local_installers/${ALPAKA_CUDA_PKG_FILE_NAME}
             ALPAKA_CUDA_OLD_KEYS=false
         else
-            echo CUDA versions other than 11.2-12.6 are not currently supported on linux!
+            echo CUDA versions other than 11.2-13.0 are not currently supported on linux!
         fi
 
         # First install the local repository.

--- a/script/job_generator/alpaka_filter.py
+++ b/script/job_generator/alpaka_filter.py
@@ -130,15 +130,15 @@ def alpaka_post_filter(row: List) -> bool:
         ):
             return False
 
-        # no released nvcc version supports Clang 19 yet (latest release was CUDA 12.6)
+        # nvcc 12.8 is the minimum requirement for host compiler Clang 19
         if row_check_version(row, HOST_COMPILER, "==", "19") and row_check_version(
             row, DEVICE_COMPILER, "<=", "12.6"
         ):
             return False
 
-        # no released nvcc version supports Clang 20 yet (latest release was CUDA 12.6)
+        # nvcc 13.0 is the minimum requirement for host compiler Clang 20
         if row_check_version(row, HOST_COMPILER, "==", "20") and row_check_version(
-            row, DEVICE_COMPILER, "<=", "12.6"
+            row, DEVICE_COMPILER, "<=", "12.9"
         ):
             return False
 

--- a/script/job_generator/versions.py
+++ b/script/job_generator/versions.py
@@ -9,7 +9,6 @@ from typeguard import typechecked
 from alpaka_job_coverage.globals import *  # pylint: disable=wildcard-import,unused-wildcard-import
 from alpaka_globals import *  # pylint: disable=wildcard-import,unused-wildcard-import
 
-
 sw_versions: Dict[str, List[str]] = {
     GCC: ["11", "12", "13"],
     CLANG: ["14", "15", "16", "17", "18", "19", "20"],
@@ -21,6 +20,8 @@ sw_versions: Dict[str, List[str]] = {
         "12.4",
         "12.5",
         "12.6",
+        "12.8",
+        "12.9",
     ],
     HIPCC: ["6.0", "6.1", "6.2", "6.3", "6.4", "7.0"],
     ICPX: ["2025.0"],

